### PR TITLE
Ask User to Share Location 

### PR DESCRIPTION
**Pull request by Chewysea**
Ask user to share location and set marker to user current location.

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,24 +1,39 @@
 import './App.css';
-import { useJsApiLoader, GoogleMap, Marker } from '@react-google-maps/api';
+import { useJsApiLoader, GoogleMap, Marker} from '@react-google-maps/api';
+import {useState, useEffect} from "react";
 
-const center = { lat: 38.89766, lng: -77.0365 };
-
-const containerStyle = { width: '100%', height: '100vh' };
+const mapStyles = {        
+  height: "100vh",
```
Comment (czahie) [inline frontend/src/App.js:40]: `height`为什么用vh而不是%?

File: frontend/src/App.js [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
   const { isLoaded } = useJsApiLoader({
-    googleMapsApiKey: process.env.REACT_APP_GOOGLE_MAP_API,
-  }); 
+    googleMapsApiKey: process.env.REACT_APP_GOOGLE_MAP_API,}); 
+
+  const [ currentPosition, setCurrentPosition ] = useState({
+    lat: 38.89766, lng: -77.0365 });
+  
+    const success = position => {
+    console.log(position.coords)
```
Comment (czahie) [inline frontend/src/App.js]: remove `console.log`

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,24 +1,39 @@
...
-const containerStyle = { width: '100%', height: '100vh' };
+const mapStyles = {        
+  height: "100vh",
+  width: "100%"};
 
 function App() {
   const { isLoaded } = useJsApiLoader({
-    googleMapsApiKey: process.env.REACT_APP_GOOGLE_MAP_API,
-  }); 
+    googleMapsApiKey: process.env.REACT_APP_GOOGLE_MAP_API,});
```
Comment (czahie) [inline frontend/src/App.js]: API key 要从secretmanager里面拿，问问Sharon怎么搞

File: frontend/src/App.js [hunk 4]
```diff
@@ -1,24 +1,39 @@
...
+    console.log(position.coords)
+    setCurrentPosition ({lat: position.coords.latitude,
+      lng: position.coords.longitude} );};
+
+  useEffect(() => {
+    navigator.geolocation.getCurrentPosition(success);
+  },[navigator.geolocation])
 
   if (isLoaded) {
+    console.log(currentPosition)
```
Comment (czahie) [inline frontend/src/App.js]: remove `console.log`

**Review by czahie — COMMENTED** [review]
整体代码格式可以稍微好一点，比如用不用分号要每一行保持一致，有没有空行也要看情况，可以查一下code style convention

File: frontend/src/App.js (hunk 1 above)
Comment (Chewysea) [reply frontend/src/App.js:40]: Based on research: A height in percentage needs a height set on the parent element as well, whereas vh doesn’t, so you can use it “everywhere”.

File: frontend/src/App.js [hunk 5]
```diff
@@ -1,24 +1,71 @@
...
   if (isLoaded) {
     return (
-      <GoogleMap zoom={10} center={center} mapContainerStyle={containerStyle}>
-        <Marker position={center} />
-      </GoogleMap>
-    );
-  } else {
+      <GoogleMap mapContainerStyle={mapStyles} zoom={13}
+      center={currentPosition}> <Marker position={currentPosition} />
+      </GoogleMap>);}
```
Comment (czahie) [inline frontend/src/App.js:63]: Minor: `zoom`里面`13`这种最好放到一个变量里面去

**Review by czahie — APPROVED** [review]

**Pull request closed without merging**